app/javascript/mastodon/components/featured_carousel.tsx

```diff
@@ -0,0 +1,182 @@
+  },
+});
+
+export const FeaturedCarousel: React.FC<{
+  accountId: string;
+  tagged?: string;
+}> = ({ accountId, tagged }) => {
+  const intl = useIntl();
+
+  // Load pinned statuses
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    if (accountId) {
+      void dispatch(expandAccountFeaturedTimeline(accountId, { tagged }));
+    }
+  }, [accountId, dispatch, tagged]);
+  const pinnedStatuses = useAppSelector(
+    (state) =>
+      (state.timelines as ImmutableMap<string, unknown>).getIn(
+        [`account:${accountId}:pinned${tagged ? `:${tagged}` : ''}`, 'items'],
+        ImmutableList(),
+      ) as ImmutableList<string>,
+  );
+
+  // Handle slide change
+  const [slideIndex, setSlideIndex] = useState(0);
+  const handleSlideChange = useCallback(
+    (direction: number) => {
+      const max = pinnedStatuses.size - 1;
+      setSlideIndex((prev) => {
+        const newIndex = prev + direction;
+        if (newIndex < 0) {
+          return max;
+        } else if (newIndex > max) {
+          return 0;
+        }
+        return newIndex;
+      });
+    },
+    [pinnedStatuses.size],
+  );
+
+  // Handle swiping animations
+  const { x } = useSpring({ x: `-${slideIndex * 100}%` });
+  const bind = useDrag(({ swipe: [swipeX] }) => {
+    handleSlideChange(swipeX * -1); // Invert swipe as swiping left loads the next slide.
+  });
+  const handlePrev = useCallback(() => {
+    handleSlideChange(-1);
+  }, [handleSlideChange]);
+  const handleNext = useCallback(() => {
+    handleSlideChange(1);
+  }, [handleSlideChange]);
+
+  // Handle wrapper height animation
+  const [wrapperStyles, wrapperSpringApi] = useSpring(() => ({
+    height: 100,
+  }));
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const wrapperEle = wrapperRef.current;
+  useEffect(() => {
+    if (!wrapperEle) {
+      return;
+    }
+    const currentSlideEle = wrapperEle.querySelector(
+      `[data-index="${slideIndex}"] > div`,
+    );
+
+    if (!currentSlideEle) {
+      return;
+    }
+    const height = currentSlideEle.scrollHeight;
+    if (!height) {
+      return;
+    }
+
+    void wrapperSpringApi.start({ height });
+  }, [slideIndex, wrapperSpringApi, wrapperEle]);
+
+  if (!accountId || pinnedStatuses.isEmpty()) {
+    return null;
+  }
+
+  return (
+    <div
+      className='featured-carousel'
+      {...bind()}
+      aria-roledescription='carousel'
+      aria-labelledby='featured-carousel-title'
+      role='region'
+    >
+      <div className='featured-carousel__header'>
+        <h4 className='featured-carousel__title' id='featured-carousel-title'>
+          {pinnedStatuses.size > 1 ? (
+            <FormattedMessage
+              id='featured_carousel.header.plural'
+              defaultMessage='Pinned Posts'
+            />
+          ) : (
+            <FormattedMessage
+              id='featured_carousel.header.single'
+              defaultMessage='Pinned Post'
+            />
+          )}
```

Comment:
Not sure whether we have rules around this, do we need to use a `count` property and the crazy `{plural, one xyz, many xyzs}` notation here? Arguably what you have here is more readable. 😅

Reply:
that uses the `FormattedPlural` component. i used that at first but it wasn't showing items in the `en.json` file so i switched to this.

Reply:
I've seen this method used in other parts of the app:

```suggestion
          <FormattedMessage
            id='featured_carousel.header'
            defaultMessage='Pinned {count, plural, one {Post} other {Posts}}'
            count={pinnedStatuses.size}
          />
```

Reply:
`count` is not a valid prop. where did you see this?